### PR TITLE
feature(generator): add client generation

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5405)

### CODE REVIEW

File: generator/integration_tests/golden/golden_client_test.cc [hunk 1]
```diff
@@ -0,0 +1,702 @@
...
+  fut = client.UpdateDatabaseDdl(request);
+  ASSERT_EQ(std::future_status::ready, fut.wait_for(std::chrono::seconds(0)));
+  db = fut.get();
+  EXPECT_STATUS_OK(db);
+  EXPECT_THAT(db->statements(), ElementsAre("-- test only: NOT SQL"));
+}
+
+TEST(GoldenClientTest, DropDatabase) {
+  auto mock = std::make_shared<MockDatabaseAdminConnection>();
+  std::string expected_database(
```
Comment (devjgm) [inline generator/integration_tests/golden/golden_client_test.cc:258]: Total nit, but we'd normally write an initialization like this as `std::string foo = "bar"`

https://abseil.io/tips/88

File: generator/internal/client_generator.cc [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+    "$client_class_name$::~$client_class_name$() = default;\n\n");
+  // clang-format on
+
+  for (google::protobuf::MethodDescriptor const& method : methods()) {
+    auto method_signature_extension =
+        method.options().GetRepeatedExtension(google::api::method_signature);
+    for (int i = 0; i < method_signature_extension.size(); ++i) {
+      std::string method_string =
+          absl::StrCat("$client_class_name$::$method_name$($method_signature",
+                       std::to_string(i), "$) {\n");
```
Comment (devjgm) [inline generator/internal/client_generator.cc:201]: No need for `std::to_string` when using `absl::StrCat`

**Review by devjgm — APPROVED** [review]
Overall, LGTM. A couple nits.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5405)* status: 0 of 14 files reviewed, 2 unresolved discussions (waiting on @devjgm)

___
*[generator/integration_tests/golden/golden_client_test.cc, line 258 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5405#-MLJOcDWuY--FNrEKklk-r2-258:-MLUKq8a0rPnJnRGwk4g:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/01e089a2503559fa9a0952df4cd1e4911313665d/generator/integration_tests/golden/golden_client_test.cc#L258)):*
<details><summary><i>Previously, devjgm (Greg Miller) wrote…</i></summary><blockquote>

Total nit, but we'd normally write an initialization like this as `std::string foo = "bar"`

https://abseil.io/tips/88
</blockquote></details>

Done.
___
*[generator/internal/client_generator.cc, line 201 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5405#-MLJOcDWuY--FNrEKkln-r2-201:-MLUKs3jEbNQBgV8NyXz:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/01e089a2503559fa9a0952df4cd1e4911313665d/generator/internal/client_generator.cc#L201)):*
<details><summary><i>Previously, devjgm (Greg Miller) wrote…</i></summary><blockquote>

No need for `std::to_string` when using `absl::StrCat`
</blockquote></details>

Done.

**Pull request merged**